gsy_framework/community_datasheet/row_converters.py

```diff
@@ -76,7 +83,48 @@ def convert(cls, row: Dict) -> Dict:
                     f"Field can't be parsed ({field_name}). {ex}"
                 ) from ex
 
-        return {field_name.strip().lower(): value}
+        return {field_name: value}
+
+    @classmethod
+    def _validate_row(cls, row: Tuple):
+        if row[0] != "Advanced Settings" and row[2] in NULL_VALUES:
+            raise CommunityDatasheetException(
+                ("Missing value for required field. " f"Row: {row}. Required field: {row[0]}.")
+            )
+
+
+class AdvancedSettingsRowConverter:
+    """Convert from the excel row to advanced settings of the community."""
+
+    @classmethod
+    def convert(cls, row: Dict) -> Dict:
+        """Convert the row using just the field name and value.
+
+        The field name is converted to snake case.
+        """
+        cls._validate_row(row)
+
+        field_name, value = row[0], row[2]  # Each row is a tuple (setting-name, legend, value)
+        field_name = field_name.strip().lower().replace(" ", "_")
+        field_name = (
+            ADVANCED_SETTINGS_FIELD_MAPPING[field_name]
+            if field_name in ADVANCED_SETTINGS_FIELD_MAPPING
```

Comment:
I would actually expand this logic and be more explicit actually. I would even include all the CDS string fields as-is, like this:
```
{"Coefficient Algorithm": "coefficient_algorithm", "Grid Fees Reduction": "grid_fees_reduction"}
```
not just for less mental hassle, but also for being able to debug in a more sane way (at the moment if you add invalid fields, to my recollection it is not immediately clear which field is incorrectly named) and also to be searchable (e.g. if I globally search for "Coefficient Algorithm" I want to find where is this CDS header defined)

Reply:
OK, i get your point.
For this I need to omit the `field_name.strip().lower()` and extend the mentioned mapping (only writing it down here in order to sync)